Server/src/controllers/mentorship.controller.js

```diff
@@ -17,122 +17,76 @@ class MentorShipController {
-      if (getMentor !== isSessionRequested.mentorEmail) {
-        return res.status(403).json({
-          status: 403,
-          error: 'Sorry, you are not allowed to accept this request'
-        });
-      } else {
-        if (isSessionRequested.status === 'accepted') {
-          return res.status(400).json({
-            status: 400,
-            error: 'Session Already accepted'
-          });
+  static async acceptMentorshipRequest(req, res) {
+    try {
+      const getMentor = req.user.email;
+      const { sessionId } = req.params;
+      const isSessionRequested = await executor(myQuery.mentorships.isSessionRequested, [sessionId]);
+      if (isSessionRequested[0]) {
+        if (getMentor !== isSessionRequested[0].mentoremail) {
+          return res.status(403).json({ status: 403, error: 'Sorry, you are not allowed to accept this request' });
         } else {
-          const result = taker.map(s => {
-            s.status = 'accepted';
-            return s;
-          });
-          return res.status(200).json({
-            status: 200,
-            message: "accepted successfully",
-            data: result
-          });
+          if (isSessionRequested[0].status === 'accepted') {
+            return res.status(400).json({ status: 409, error: 'Session Already accepted' });
+          } else {
+            const result = await executor(myQuery.mentorships.acceptSession, [sessionId]);
+            return res.status(200).json({ status: 200, message: "accepted successfully", data: result[0] });
+          }
         }
       }
+      return res.status(404).json({ status: 404, error: "Session not found" });
+    } catch (err) {
+      return res.status(400).json({ status: 400, error: err.message });
     }
-    return res.status(404).json({
-      status: 404,
-      error: "Session not found"
-    });
-
   }
   /**
   * Reject mentorship session request
   * @param {object} req
   * @param {object} res
   */
-  static rejectMentorshipRequest(req, res) {
-    const getMentor = req.user.email;
-    let taker = session;
-    const isSessionRequested = taker.find(s => s.sessionId == parseInt(req.params.sessionId));
-    taker = new Array(isSessionRequested);
-    if (isSessionRequested) {
-      if (getMentor !== isSessionRequested.mentorEmail) {
-        return res.status(403).json({
-          status: 403,
-          error: 'Sorry, you are not allowed to reject this request'
-        });
-      } else {
-        if (isSessionRequested.status === 'rejected') {
-          return res.status(400).json({
-            status: 400,
-            error: 'Session Already rejected'
-          });
+  static async rejectMentorshipRequest(req, res) {
+    try {
+      const getMentor = req.user.email;
+      const { sessionId } = req.params;
+      const isSessionRequested = await executor(myQuery.mentorships.isSessionRequested, [sessionId]);
+      if (isSessionRequested[0]) {
+        if (getMentor !== isSessionRequested[0].mentoremail) {
```

Comment:
This should be handled by the middleware, you need to move this code to the middleware